tests/test_requirement.py

```diff
@@ -1,781 +1,867 @@
+    ('2', '>', '2.4.8.post1', False),
+    ('2.0.0b5', '>', '2.4.8.post1', False),
+    ('2.0.post1', '>', '2.4.8.post1', False),
+    ('2.4.8', '>', '2.0', True),
+    ('2.4.8.0', '>', '2.0', True),
+    ('2.4.8.1', '>', '2.0', True),
+    ('2.4.8b5', '>', '2.0', True),
+    ('2.4.8.post1', '>', '2.0', True),
+    ('2.0', '>', '2.0', False),
+    ('2', '>', '2.0', False),
+    ('2.0.0b5', '>', '2.0', False),
+    ('2.0.post1', '>', '2.0', False),
+    ('2.4.8', '>', '2', True),
+    ('2.4.8.0', '>', '2', True),
+    ('2.4.8.1', '>', '2', True),
+    ('2.4.8b5', '>', '2', True),
+    ('2.4.8.post1', '>', '2', True),
+    ('2.0', '>', '2', False),
+    ('2', '>', '2', False),
+    ('2.0.0b5', '>', '2', False),
+    ('2.0.post1', '>', '2', False),
+    ('2.4.8', '>', '2.*', True),
+    ('2.4.8.0', '>', '2.*', True),
+    ('2.4.8.1', '>', '2.*', True),
+    ('2.4.8b5', '>', '2.*', True),
+    ('2.4.8.post1', '>', '2.*', True),
+    ('2.0', '>', '2.*', True),
+    ('2', '>', '2.*', True),
+    ('2.0.0b5', '>', '2.*', True),
+    ('2.0.post1', '>', '2.*', True),
+    ('2.4.8', '>', '2.0.0b5', True),
+    ('2.4.8.0', '>', '2.0.0b5', True),
+    ('2.4.8.1', '>', '2.0.0b5', True),
+    ('2.4.8b5', '>', '2.0.0b5', True),
+    ('2.4.8.post1', '>', '2.0.0b5', True),
+    ('2.0', '>', '2.0.0b5', True),
+    ('2', '>', '2.0.0b5', True),
+    ('2.0.0b5', '>', '2.0.0b5', False),
+    ('2.0.post1', '>', '2.0.0b5', False),
+    ('2.4.8', '>', '2.0.post1', True),
+    ('2.4.8.0', '>', '2.0.post1', True),
+    ('2.4.8.1', '>', '2.0.post1', True),
+    ('2.4.8b5', '>', '2.0.post1', True),
+    ('2.4.8.post1', '>', '2.0.post1', True),
+    ('2.0', '>', '2.0.post1', False),
+    ('2', '>', '2.0.post1', False),
+    ('2.0.0b5', '>', '2.0.post1', False),
+    ('2.0.post1', '>', '2.0.post1', False),
+    ('1.dev1', '>', '1a1', False),
+    ('1a1', '>', '1.dev1', True),
+    ('1a1', '>', '1b1', False),
+    ('1b1', '>', '1a1', True),
+    ('1b1', '>', '1rc1', False),
+    ('1', '>', '1rc1', True),
+    ('1', '>', '1.post1', False),
+    ('1.post1', '>', '1', False),
 ])
-def test_requirement(version, arg, expected):
-    assert (version in pkg_resources.Requirement.parse(arg)) == expected
+def test_requirement(version, op, arg, expected):
+    assert (version in pkg_resources.Requirement.parse('foobar {} {}'.format(op, arg))) == expected
+    # The section below will attempt further validation of accurate conversion using
+    # rpmdev-vercmp.  However, that can only tell us that two versions are equal or
+    # that one is greater than the other.  We can't test rich deps.
+    requirement = convert('foobar', op, arg)
+    if op == '===':
+        # We treat '===' as '==', which is not entirely the same behavior
+        return
+    if requirement[0] == '(':
+        # ~= and != will produce rich deps, which we can't currently test.
+        return
+
+    ver_a = RpmVersion(version)
+    (rpm_op, ver_b) = requirement.split(' ')[1:3]
```

Comment:
A nit pick. I consider this way of writing it more common:

```suggestion
    rpm_op, ver_b = requirement.split(' ')[1:3]
```

Feel free to disregard this.